File: Dataset/Leetcode/train/14/41.py
 class Solution:
    def XXX(self, strs: List[str]) -> str:
        if strs == []:
            return ''
        out = strs[0]
        for i in strs:
            while out not in i or i.index(out) != 0:
                out = out[:-1]
                if out == '':
                    return ''

        return out

File: Dataset/Leetcode/test/19/41.cpp
 class Solution {
public:
    ListNode* XXX(ListNode* head, int n) {
        ListNode* preNode = new ListNode();
        preNode->next = head;
        ListNode* fast = preNode, *slow = preNode, *pre = preNode;
        for(int i = 0; i < n - 1; i++)
        {
            fast = fast->next;
        }
        while(fast->next)
        {   if(slow != preNode)
                pre = pre->next;
            fast = fast->next;
            slow = slow->next;
        }
        pre->next = pre->next->next;
        return preNode->next;
    }
};

